Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -157,11 +165,33 @@ open class ExtendedJSONConversionTestCase: BSONTestCase {
     }
 
     func testCodeWScope() throws {
-        // Skipping until there is a BSON.init(fromExtJSON) because its recursive
+        // Success case
+        expect(try BSONCodeWithScope(fromExtJSON: ["$code": "javascript", "$scope": ["doc": "scope"]], keyPath: []))
+            .to(equal(BSONCodeWithScope(
+                code: "javascript",
+                scope: BSONDocument(keyValuePairs: [("doc", BSON.string("scope"))])
+            )))
+
+        // Error cases
+        expect(try BSONCodeWithScope(fromExtJSON: ["$code": 5, "$scope": ["doc": "scope"]], keyPath: []))
+            .to(throwError(errorType: DecodingError.self))
+        expect(try BSONCodeWithScope(fromExtJSON: ["$code": "javascript", "$scope": 1], keyPath: []))
+            .to(throwError(errorType: DecodingError.self))
     }
 
     func testDocument() throws {
-        // Skipping until there is a BSON.init(fromExtJSON) because its recursive
+        // Success case
+        expect(try BSONDocument(fromExtJSON: ["key": ["$numberInt": "5"]], keyPath: []))
+            .to(equal(BSONDocument(keyValuePairs: [("key", .int32(5))])))
+        // Nil case
+        expect(try BSONDocument(fromExtJSON: 1, keyPath: [])).to(beNil())
+
+        // Error case
+        expect { try BSONDocument(fromExtJSON: ["time": ["$timestamp": 5]], keyPath: []) }
+            .to(throwError(DecodingError._extendedJSONError(
```

Comment:
This test is a little too specific to the implementation, which means it will need to be updated if we ever change the format of the error message. The key things we do need to test here are that the error is in fact a `DecodingError` and that the keypath shows up properly. To do so, you can use that nimble matcher I showed you and do some checks on the error:

```swift
expect(try BSONDocument(...)).to(throwError { error in
    expect(error).to(matchError(DecodingError.self))
    expect(error.localizedDescription).to(contain("time"))
})
```
This is a little more resilient to implementation changes while still verifying the behavior we want.

Reply:
I tried using exactly the code you suggested, but `error.localizedDescription` returns "The data couldn’t be read because it isn’t in the correct format." somehow I need to get at the `debugDescription` in the context of `DecodingError.dataCorrupted`, but I cant figure out a way to do so.

Reply:
Discussed offline: can use `String(describing:)` here.